integration/marathon_test.go

```diff
@@ -48,6 +75,60 @@ func (s *MarathonSuite) TestSimpleConfiguration(c *check.C) {
 	// TODO validate : run on 80
 	// Expected a 404 as we did not configure anything
 	err = try.GetRequest("http://127.0.0.1:8000/", 500*time.Millisecond, try.StatusCodeIs(http.StatusNotFound))
+	c.Assert(err, checker.IsNil)
+}
+
+func (s *MarathonSuite) TestConfigurationUpdate(c *check.C) {
+	marathonURL := "http://" + s.getContainerIPAddr(c, containerNameMarathon) + ":8080"
+	fmt.Printf("Using Marathon URL %s\n", marathonURL)
+
+	// Wait for Marathon readiness prior to creating the client so that we
+	// don't run into the "all cluster members down" state right from the
+	// start.
+	fmt.Println("Waiting for Marathon to become ready")
+	err := try.GetRequest(marathonURL+"/ping", 1*time.Minute, try.StatusCodeIs(http.StatusOK))
+	c.Assert(err, checker.IsNil)
+
+	file := s.adaptFile(c, "fixtures/marathon/simple.toml", struct {
+		MarathonURL string
+	}{marathonURL})
+	defer os.Remove(file)
+	cmd, output := s.cmdTraefik(withConfigFile(file))
+	err = cmd.Start()
+	c.Assert(err, checker.IsNil)
+	defer cmd.Process.Kill()
+
+	// Prepare Marathon client.
+	config := marathon.NewDefaultConfig()
+	config.URL = marathonURL
+	client, err := marathon.NewClient(config)
+	c.Assert(err, checker.IsNil)
+
+	showTraefikLog := true
```

Comment:
See the last line of the test method: I flip the flag if the test runs to completion successfully so that we don't show the logs if everything went fine.